### ISSUE TITLE
Add text model metrics to HGCN validation (cophenetic correlation, NDCG, etc.)

### ISSUE THREAD

**Issue by lowmason** [lowmason/naics-embedder#70]
## Summary

The HGCN (Stage 4) validation currently uses a limited set of metrics compared to the text model stages. Adding the richer metrics from Stages 1–3 would help assess whether HGCN maintains global structure while improving local relations.

## Current State

**Text model (Stages 1–3) includes:**
- Cophenetic correlation
- NDCG (Normalized Discounted Cumulative Gain)
- Global structure preservation metrics
- Ranking quality measures

**HGCN (Stage 4) currently tracks:**
- Triplet loss
- Relation accuracy (per-batch)
- Radius mean/std
- Basic distance metrics

## Suggested Addition

Port the following metrics to `HGCNLightningModule` validation:
```python
def validation_step(self, batch, batch_idx):
    # Existing metrics
    loss = self._compute_triplet_loss(...)
    self.log('val/loss', loss)
    
    # Add text model metrics
    if self.global_step % self.hparams.full_eval_frequency == 0:
        cophenetic_corr = compute_cophenetic_correlation(
            embeddings,
            hierarchy,
        )
        ndcg = compute_ndcg(embeddings, hierarchy, k=10)
        
        self.log('val/cophenetic_correlation', cophenetic_corr)
        self.log('val/ndcg@10', ndcg)
```

## Why This Matters

HGCN fine-tunes embeddings to improve parent–child locality, but this could inadvertently distort global structure. Without the same metrics used in earlier stages, we can't verify that Stage 4 preserves what Stages 1–3 achieved.

## Priority

Medium—important for validating end-to-end pipeline quality.
